contrib/externs/chrome_extensions.js

```diff
@@ -11214,3 +11214,96 @@ chrome.downloads.DeterminingFilenameEvent.prototype.hasListeners =
  * https://developer.chrome.com/extensions/downloads#event-onDeterminingFilename
  */
 chrome.downloads.onDeterminingFilename;
+
+
+/**
+ * @const
+ * @see https://developer.chrome.com/extensions/declarativeContent
+ */
+chrome.declarativeContent = {};
+
+
+/** @type {string} */
+chrome.declarativeContent.PageStateMatcherInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.ShowPageActionInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.SetIconInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.RequestContentScriptInstanceType;
+
+
+/**
+ * @typedef {{
+ *   pageUrl: (!UrlFilter|undefined),
+ *   css: (!Array<string>|undefined),
+ *   isBookmarked: (boolean|undefined)
+ * }}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher
+ */
+chrome.declarativeContent.PageStateMatcherLiteral;
+
+
+/**
+ * @constructor
+ * @param {?chrome.declarativeContent.PageStateMatcherLiteral=} literalValue
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher
+ */
+chrome.declarativeContent.PageStateMatcher = function(literalValue) {};
+
+
+/**
+ * @constructor
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-ShowPageAction
+ */
+chrome.declarativeContent.ShowPageAction = function() {};
+
+
+/**
+ * @typedef {{
+ *   imageData: (!ImageData|!Object<number, !ImageData>|undefined)
+ * }}
+ * @see https://developer.chrome.com/extensions/declarativeContent#type-SetIcon
+ */
+chrome.declarativeContent.SetIconLiteral;
+
+
+/**
+ * @constructor
+ * @param {?chrome.declarativeContent.SetIconLiteral=} literalValue
```

Comment:
I suspect that if you actually pass a `null` value as the argument for this constructor and the similar ones below you'll get an exception.

Could you please confirm that passing `null` doesn't break things or change the `?` here and below to `!`?

Thanks.

Reply:
@brad4d I tested passing `null` to `PageStateMatcher` and no error was thrown.
Minified version result: `new chrome.declarativeContent.PageStateMatcher(null)`.

When passing `null` to `RequestContentScript` it throws:
`Unchecked runtime.lastError while running events.addRules: Missing parameter is required: css or js`.
So, docs say they are optional parameters but at least one is needed.

When passing `null` to `SetIcon` it throws:
`Error in event handler for runtime.onInstalled: TypeError: Cannot use 'in' operator to search for 'iconIndex' in null`

Reply:
@brad4d based on the errors thrown, should I change the `?` to `!`?

Reply:
@IvanRF yes, please change `?` to `!` for methods where passing `null` causes an exception.
Thanks.

Reply:
@brad4d done